src/components/CreateMatchExperienceModal/index.tsx

```diff
@@ -74,22 +78,72 @@ const CreateMatchExperienceModal = ({ isOpen, onClose }: CreateMatchExperienceMo
         },
     });
 
+    const handleYear = (date: Date): number => {
+        return date.getMonth() >= 6 ? date.getFullYear() : date.getFullYear() - 1;
+    };
+
     const { data: countries = [] } = useQueryOnDefinedParam('countries', isOpen, SoccerService.getCountries);
-    const { data: leagues = [] } = useQueryOnDefinedParam('leagues', selectedCountry, SoccerService.getLeagues);
-    const { data: stadiums = [] } = useQueryOnDefinedParam('stadiums', selectedCountry, SoccerService.getVenues);
-    const { data: teams = [] } = useQueryOnDefinedParam('teams', selectedLeague, SoccerService.getTeams);
+    const { data: leagues = [] } = useQueryOnDefinedParam(
+        'leagues',
+        selectedCountry ? selectedCountry : undefined,
+        SoccerService.getLeagues
+    );
+    const { data: stadiums = [] } = useQueryOnDefinedParam(
+        'stadiums',
+        selectedCountry ? selectedCountry : undefined,
+        SoccerService.getVenues
+    );
 
-    const {
-        control,
-        handleSubmit,
-        setValue,
-        reset,
-        formState: { errors },
-    } = useForm<CreateMatchExperienceModalValues>({
-        resolver: yupResolver(MatchExperienceSchema),
-    });
+    useEffect(() => {
+        if (selectedLeague !== undefined && selectedDate !== null) {
+            const season = handleYear(new Date(selectedDate));
+            SoccerService.getTeams({ leagueId: selectedLeague, season }).then((data) => setTeams(data));
+        }
+    }, [selectedLeague, selectedDate]);
+
+    const resetValuesOnCountryChange = (value: string) => {
+        setSelectedCountry(value);
+        setValue('country', value);
+        setValue('league', '');
+        setValue('stadium', '');
+        setValue('homeTeam', '');
+        setValue('awayTeam', '');
```

Comment:
```suggestion
      reset((prevValues) => ({
        ...prevValues, // Keep other form values unchanged
        league: value,
        homeTeam: '',
        awayTeam: '',
    }));
```

use reset function of `useForm` for batch form values change. same on other cases

Reply:
Why do I need this? This function is meant to reset specific fields, while the ones that I didn't change remain the same.

Reply:
The suggestion I gave does the same as the function did before just with cleaner syntax. you are doing one batch update for multiple fields instead just for one

Reply:
i never need `prevValues` , because i always update all of them ( it reset the values i need )